app/inputexpressionfunctions.h

```diff
@@ -0,0 +1,172 @@
+# else
+          return QString();
+#endif
+        }
+
+        QgsScopedExpressionFunction *clone() const override { return new ReadExifImgDirection();}
+    };
+
+
+    //! Expression function for reading EXIF Latitude
+    class ReadExifLatitude : public QgsScopedExpressionFunction
+    {
+      public:
+        ReadExifLatitude()
+          : QgsScopedExpressionFunction( QStringLiteral( "read_exif_latitude" ), QgsExpressionFunction::ParameterList()
+                                         << QgsExpressionFunction::Parameter( QStringLiteral( "file" ) ),
+                                         QStringLiteral( "Custom" ) ) {}
+        /**
+         * @param values - contain one parameter:
+         *  - file: Absolute path of an image that exif attribute will be read from,
+         * @return EXIF value for GPS Latitude
+         */
+        QVariant func( const QVariantList &values, const QgsExpressionContext *, QgsExpression *, const QgsExpressionNodeFunction * ) override
+        {
+#ifdef ANDROID
+          if ( values.size() != 1 ) return QVariant();
+
+          QString filepath( values.at( 0 ).toString() );
+          QString resultString = AndroidUtils::getExifInfo( filepath, GPS_LAT_TAG );
+          if ( resultString.isEmpty() )
+            return QVariant();
+
+          return QVariant( InputUtils::convertCoordinateString( resultString ) );
+# else
+          return QString();
+#endif
+        }
+
+        QgsScopedExpressionFunction *clone() const override { return new ReadExifLatitude();}
+    };
+
+
+    //! Expression function for reading EXIF Longitude
+    class ReadExifLongitude : public QgsScopedExpressionFunction
+    {
+      public:
+        ReadExifLongitude()
+          : QgsScopedExpressionFunction( QStringLiteral( "read_exif_longitude" ), QgsExpressionFunction::ParameterList()
+                                         << QgsExpressionFunction::Parameter( QStringLiteral( "file" ) ),
+                                         QStringLiteral( "Custom" ) ) {}
+        /**
+         * @param values - contain one parameter:
+         *  - file: Absolute path of an image that exif attribute will be read from,
+         * @return EXIF value for GPS Longitude
+         */
+        QVariant func( const QVariantList &values, const QgsExpressionContext *, QgsExpression *, const QgsExpressionNodeFunction * ) override
+        {
+#ifdef ANDROID
+          if ( values.size() != 1 ) return QVariant();
+
+          QString filepath( values.at( 0 ).toString() );
+          QString resultString = AndroidUtils::getExifInfo( filepath, GPS_LON_TAG );
+          if ( resultString.isEmpty() )
+            return QVariant();
+
+          return QVariant( InputUtils::convertCoordinateString( resultString ) );
+# else
+          return QString();
+#endif
+        }
+
+        QgsScopedExpressionFunction *clone() const override { return new ReadExifLongitude();}
+    };
+
+    InputExpressionFunctions();
```

Comment:
why we need custom constructor, if needed, move to the top of the file also with the register Functions function